Client/Frontend/Autofill/CreditCard/CreditCardBottomSheet/CreditCardBottomSheetViewModel.swift

```diff
@@ -161,25 +179,61 @@ struct CreditCardBottomSheetViewModel {
                                                                        with: ccNumberDecrypted,
                                                                        fieldValues: decryptedCreditCard)
             return updatedDecryptedCreditCard
+        case .selectSavedCard:
+            guard let row = row,
+                  let selectedCreditCard = getSavedCreditCard(for: row)
+            else {
+                return nil
+            }
+
+            let decryptedCreditCardNum = decryptCreditCardNumber(card: selectedCreditCard)
+
+            guard !decryptedCreditCardNum.isEmpty else {
+                return nil
+            }
+
+            let plainTextCard = UnencryptedCreditCardFields(ccName: selectedCreditCard.ccName,
+                                                            ccNumber: decryptedCreditCardNum,
+                                                            ccNumberLast4: selectedCreditCard.ccNumberLast4,
+                                                            ccExpMonth: selectedCreditCard.ccExpMonth,
+                                                            ccExpYear: selectedCreditCard.ccExpYear,
+                                                            ccType: selectedCreditCard.ccType)
+            return plainTextCard
         }
     }
 
     func getConvertedCreditCardValues(bottomSheetState: CreditCardBottomSheetState,
-                                      ccNumberDecrypted: String) -> CreditCard? {
+                                      ccNumberDecrypted: String,
+                                      row: Int? = nil) -> CreditCard? {
         switch bottomSheetState {
         case .save:
             guard let plainCard = decryptedCreditCard else { return nil }
             return plainCard.convertToTempCreditCard()
         case .update:
-            guard let creditCard = creditCard, !ccNumberDecrypted.isEmpty
+            guard let creditCard = creditCard, !ccNumberDecrypted.isEmpty,
+                  let updatedDecryptedCreditCard = updateDecryptedCreditCard(
+                    from: creditCard,
+                    with: ccNumberDecrypted,
+                    fieldValues: decryptedCreditCard)
             else {
                 return nil
             }
-            let updatedDecryptedCreditCard = updateDecryptedCreditCard(from: creditCard,
-                                                                       with: ccNumberDecrypted,
-                                                                       fieldValues: decryptedCreditCard)
-            return updatedDecryptedCreditCard?.convertToTempCreditCard()
+            return updatedDecryptedCreditCard.convertToTempCreditCard()
+        case .selectSavedCard:
+            guard let row = row else { return nil }
+            return getSavedCreditCard(for: row)
+        }
+    }
+
+    private func getSavedCreditCard(for row: Int) -> CreditCard? {
+        guard row > -1,
```

Comment:
Curious under which conditions row could be negative? and in case is possible a Unit test will be perfect to cover that edge case

Reply:
It can probably be a programmer error but because we rely on index paths from tableview we should not have this issue. I added a test for that called `test_update_selectConvertedCreditCardValues_ForMinusRow`.